docs/collision-detection.md

```diff
@@ -0,0 +1,54 @@
+# Collision Detection 💥
+
+Flame comes with a powerful built-in [collision detection system](https://docs.flame-engine.org/1.6.0/flame/collision_detection.html),
+but this API is not strongly typed. Components always get the colliding component as a
+`PositionComponent` and developers need to manually check what type of class it is.
+
+`flame_behaviors` is all about enforcing a strongly typed API. It provides a special behavior
+called `CollisionBehavior` that describes what type of entity it will target for collision. It
+does not, however, do any real collision detection. That is done by the
+`PropagatingCollisionBehavior`.
+
+The `PropagatingCollisionBehavior` handles the collision detection by registering a hitbox on the
+parent entity. When that hitbox has a collision, the `PropagatingCollisionBehavior` checks if the
+component that the parent entity is colliding with contains the target entity type specified in
+`CollisionBehavior`.
+
+By letting the `PropagatingCollisionBehavior` handle the collision detection we gain two benefits,
+the first and most important one is performance. By only registering collision callbacks on the
+entities themselves, the collision detection system does not have to go through any "collidable"
+behaviors, for which there could be many per entity. We only do that now if we confirm a collision
+has happened.
+
+The second benefit is that it allows for [separation of concerns][separation_of_concerns].
```

Comment:
```suggestion
The second benefit is that it allows for [separation of concerns][separation_of_concerns].
```

Aren't we missing the link for `separation_of_concerns` here?

Reply:
👀  maybe? I'll check up what it was originally

Reply:
Will be added in next commit!